first_order_model/shrink_util.py

```diff
@@ -588,18 +636,33 @@ def add_names(names):
             'up_blocks.0.norm', 'up_blocks.1.conv.depth_conv',
             'up_blocks.1.conv.point_conv', 'up_blocks.1.norm',
             'hr_up_blocks.0.conv.depth_conv', 'hr_up_blocks.0.conv.point_conv',
-            'hr_up_blocks.0.norm', 'final.depth_conv', 'final.point_conv'
-        ])
+            'hr_up_blocks.0.norm'])
+
+        if is_1024:
+            add_names(['hr_up_blocks.0.norm', 'hr_up_blocks.1.conv.depth_conv', 'hr_up_blocks.1.conv.point_conv', 'hr_up_blocks.1.norm', 'final.depth_conv', 'final.point_conv'])
+        else:
+            add_names(['hr_up_blocks.0.norm', 'final.depth_conv', 'final.point_conv'])
 
         # Features get concatted into down block
         add('down_blocks.1.norm', 'bottleneck.r0.norm1')
 
-        # Second up block has 32 lr features added
-        add('lr_first.norm', 'up_blocks.0.conv.depth_conv')
-
-        # Add 2x hr down outputs to first hr up
-        add('hr_down_blocks.0.norm', 'hr_up_blocks.0.conv.depth_conv')
-        add('hr_down_blocks.0.norm', 'hr_up_blocks.0.conv.depth_conv')
+        if is_1024:
+            # Second up block has 32 lr features added
+            add('lr_first.norm', 'up_blocks.1.conv.depth_conv')
+        else:
+            # Second up block has 32 lr features added
+            add('lr_first.norm', 'up_blocks.0.conv.depth_conv')
+
+        if is_1024:
+            # Add 2x hr down outputs to first hr up
+            add('hr_down_blocks.1.norm', 'hr_up_blocks.0.conv.depth_conv')
```

Comment:
I moved the comments out of the block, other than that this part is fairly hard to comment as its just manually following the codebase and looking at all the connections. 512 and 1024 are different since they have slightly different skip connection structures (see the 1 and 0's in the block are slightly different for both cases).

Reply:
Does this make more sense now?